File: python_leaoai/python_leaoai/repositories/Factories/Catalogo/EquipeFactory.py
from typing import List, Optional
from sqlalchemy.orm import Session
from sqlalchemy.exc import SQLAlchemyError
from models.catalogo import Equipe  # Importe seu modelo aqui

class EquipeFactory:
    def __init__(self, db: Session):
        self.db = db

    # -------------------------
    # CRUD Síncrono
    # -------------------------
    def get_obj(self, id: int) -> Optional[Equipe]:
        return self.db.query(Equipe).filter(Equipe.id == id).first()

    def save_obj(self, obj: Equipe) -> Equipe:
        self.db.add(obj)
        self.db.commit()
        self.db.refresh(obj)
        return obj

    def update_obj(self, obj: Equipe) -> Equipe:
        self.db.add(obj)
        self.db.commit()
        self.db.refresh(obj)
        return obj

    def get_all(self) -> List[Equipe]:
        return self.db.query(Equipe).all()

    def get_list(self) -> List[Equipe]:
        return self.db.query(Equipe).filter(Equipe.excluido != True).all()

    def get_all_ativo(self) -> List[Equipe]:
        return self.db.query(Equipe).filter(Equipe.ativo == True).all()

    def get_all_excluido(self) -> List[Equipe]:
        return self.db.query(Equipe).filter(Equipe.excluido == True).all()

    def remove_obj(self, obj: Equipe) -> bool:
        try:
            self.db.delete(obj)
            self.db.commit()
            return True
        except SQLAlchemyError:
            self.db.rollback()
            return False

    def delete_obj(self, id: int) -> bool:
        obj = self.get_obj(id)
        if not obj:
            return False
        try:
            obj.ativo = False
            obj.excluido = True
            self.db.add(obj)
            self.db.commit()
            return True
        except SQLAlchemyError:
            self.db.rollback()
            return False

    def get_by_id(self, equipe_id: int) -> Equipe:
        equipe = self.get_obj(equipe_id)
        if not equipe:
            raise Exception(f"Equipe com ID {equipe_id} não encontrada.")
        return equipe

    # -------------------------
    # Métodos específicos para imagens
    # -------------------------
    def get_all_to_blob(self) -> List[Equipe]:
        return self.db.query(Equipe).filter(
            ((Equipe.imagem != None) & (Equipe.imagem != "") & (~Equipe.imagem.contains("http"))) |
            ((Equipe.thumbnail != None) & (Equipe.thumbnail != "") & (~Equipe.thumbnail.contains("http")))
        ).all()

    def get_all_to_cdn(self) -> List[Equipe]:
        return self.db.query(Equipe).filter(
            ((Equipe.imagem != None) & (Equipe.imagem != "") & (Equipe.imagem.contains("http")) & (~Equipe.imagem.startswith("https://cdn.codiehost.com.br/"))) |
            ((Equipe.thumbnail != None) & (Equipe.thumbnail != "") & (Equipe.thumbnail.contains("http")) & (~Equipe.thumbnail.startswith("https://cdn.codiehost.com.br/")))
        ).all()

    # -------------------------
    # Async update (exemplo)
    # -------------------------
    async def update_obj_async(self, obj: Equipe, async_db):
        async_db.add(obj)
        await async_db.commit()
        await async_db.refresh(obj)
        return obj

File: Framework.Repositories/Factories/Catalogo/EquipeFactory.cs
using Data;
using System.Linq;
using Data.Models.Catalogo;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Framework.Factories.Catalogo
{
    public class EquipeFactory
    {
        private readonly ApplicationDbContext _context;

        public EquipeFactory(ApplicationDbContext context)
        {
            _context = context;
        }

        #region Crud
        public Equipe GetObj(int id)
        {
            return _context.Equipe.Where(b => b.Id == id).FirstOrDefault();
        }
        public Equipe SaveObj(Equipe obj)
        {
            _context.Add(obj);
            _context.SaveChanges();

            return obj;
        }
        public Equipe UpdateObj(Equipe obj)
        {
            _context.Update(obj);
            _context.SaveChanges();
            return obj;
        }
        public List<Equipe> GetAll()
        {
            return _context.Equipe.ToList();
        }
        public List<Equipe> GetList()
        {
            return _context.Equipe.Where(b => b.Excluido != true).ToList();
        }
        public List<Equipe> GetAllAtivo()
        {
            return _context.Equipe.Where(b => b.Ativo == true).ToList();
        }
        public List<Equipe> GetAllExcluido()
        {
            return _context.Equipe.Where(b => b.Excluido == true).ToList();
        }
        public Boolean RemoveObj(Equipe obj)
        {
            try
            {
                _context.Equipe.Remove(obj);
                _context.SaveChanges();

                return true;
            }
            catch (Exception)
            {
                return false;
            }
        }
        public Boolean DeleteObj(int Id)
        {
            try
            {
                var obj = _context.Equipe.Where(b => b.Id == Id).FirstOrDefault();

                obj.Ativo = false;
                obj.Excluido = true;

                _context.Equipe.Update(obj);
                _context.SaveChanges();

                return true;
            }
            catch (Exception)
            {
                return false;
            }
        }

        public Equipe GetByID(int equipeID)
        {
            // Suponha que "Equipe" seja o nome da sua entidade no contexto do banco de dados
            Equipe equipe = _context.Equipe.FirstOrDefault(e => e.Id == equipeID);

            if (equipe == null)
            {
                throw new Exception($"Equipe com ID {equipeID} não encontrada.");
            }

            return equipe;
        }

        public List<Equipe> GetAllToBlob()
        {
            return _context.Equipe
                .Where(x => (!string.IsNullOrWhiteSpace(x.Imagem) && !x.Imagem.Contains("http"))
                || (!string.IsNullOrWhiteSpace(x.Thumbnail) && !x.Thumbnail.Contains("http"))
                )
                .ToList();
        }

        public async Task<Equipe> UpdateObjAsync(Equipe obj)
        {
            _context.Update(obj);
            _context.SaveChanges();

            return obj;
        }

		public List<Equipe> GetAllToCDN()
		{
			return _context.Equipe
				.Where(x =>
			(!string.IsNullOrWhiteSpace(x.Imagem) && x.Imagem.Contains("http") && !x.Imagem.StartsWith("https://cdn.codiehost.com.br/")) ||
			(!string.IsNullOrWhiteSpace(x.Thumbnail) && x.Thumbnail.Contains("http")! && x.Thumbnail.StartsWith("https://cdn.codiehost.com.br/"))
			)
				.ToList();
		}
		#endregion
	}
}
